tools/readme-catalog/index.js

```diff
@@ -0,0 +1,75 @@
+const fs = require('fs');
+
+/**
+ * Generates a catalog for the README.
+ */
+const CATALOG = JSON.parse(fs.readFileSync(`${__dirname}/../../jsonschema/catalog.json`));
+
+/**
+ * Replaces the contents of a string's GENERATED comments with a replacement.
+ * @param {string} s The string to replace.
+ * @param {string} replacement The replacement string.
+ */
+const getGeneratedStringWithReplacement = (s, replacement) => {
+  const README_GEN_START = '<!-- GENERATED START -->';
+  const README_GEN_END = '<!-- GENERATED END -->';
+
+  const README_BEFORE_TABLE = s.substring(0, s.indexOf(README_GEN_START) + README_GEN_START.length);
+  const README_AFTER_TABLE = s.substring(s.indexOf(README_GEN_END));
+
+  // Return result (with newlines)
+  return `${README_BEFORE_TABLE}
+${replacement}
+${README_AFTER_TABLE}`;
+};
+
+/**
+ * Gets markdown for the proto / JSON schema table cell.
+ * @param {object} schemaEntry The JSON schema catalog entry.
+ * @returns {string} Markdown for the schema entry.
+ */
+const getSchemaLinkMarkdown = (schemaEntry) => {
+  // Get the path the to the data from the datatype.
+  // in: "google.events.cloud.audit.v1.LogEntryData"
+  // out: "cloud/audit/v1"
+  const prefix = 'google.';
+  const schemaPath = schemaEntry.datatype.substring(
+    schemaEntry.datatype.indexOf(prefix) + prefix.length,
+    schemaEntry.datatype.indexOf(`.${schemaEntry.name}`),
+  ).replace(/\./g, '/');
+
+  // Proto link is the path + data.proto
+  const protoLink = `proto/google/events/${schemaPath}/data.proto`;
```

Comment:
For the proto link, I'd expect you to be able to just remove the final part (e.g. LogEntryData), replace all dots with slashes, and add data.proto. I don't think you need to do anything a specific prefix etc.

I haven't quite followed whether the layout of the JSON schema has changed, but it looks like you don't need any prefixing for the JSON schema either - just replace "." with "/" and add ".json" at the end, and the link should work, I think...

Reply:
Some parts of this comment are correct. The `proto/` prefix is still needed though. The JSON schema does not use `data.json` (that's naming is not really common for JSON / JSON Schema).

This should be fixed, so resolving. Perhaps the logic can be further simplified, but that can be a separate refactoring if possible.